### PR TITLE
Add SwAV prototype freezing for first n steps 

### PR DESCRIPTION
**Pull request by ibro45**
https://github.com/facebookresearch/swav/blob/5e073db0cc69dea22aa75e92bfdd75011e888f28/main_swav.py#L325

### CODE REVIEW

**Review by MalteEbner — COMMENTED** [review]
Thanks a lot for finding this missing argument and your contribution!
Your implementation looks fine and working.

However, I would prefer to see the `num_steps_frozen_prototypes` as attribute of the [SwaVPrototypes](https://github.com/lightly-ai/lightly/blob/c80948773b8864f48b4fddfe757a754c20859193/lightly/models/modules/heads.py#L370) class instead of the `SwaV` class. This would
- Make the codebase more modular, e.g. you could use SwaV prototypes with other base models.
- Be consistent with the [DINOProjectionHead. freeze_last_layer](https://github.com/lightly-ai/lightly/blob/c80948773b8864f48b4fddfe757a754c20859193/lightly/models/modules/heads.py#L433-L436) argument, which has a very similar purpose. 

Then the main training loop can just call
`model.prototypes.freeze_if_required(step)` before the forward step, again consistent with [how it is done for DINO](https://github.com/lightly-ai/lightly/blob/c80948773b8864f48b4fddfe757a754c20859193/examples/pytorch/dino.py#L92)

**Comment by ibro45** [comment]
Completely agree, I pushed the changes

File: lightly/models/modules/heads.py [hunk 1]
```diff
@@ -374,6 +374,14 @@ class SwaVPrototypes(nn.Module):
...
     predict the assigned prototypes of the other augmentation.
 
+    Attributes:
+        input_dim:
+            The input dimension of the head.
+        n_prototypes:
+            Number of prototypes.
+        n_steps_frozen_prototypes:
+            Number of steps during which we keep the prototypes fixed.
+
```
Comment (MalteEbner) [inline lightly/models/modules/heads.py:384]: Nice!

**Review by MalteEbner — APPROVED** [review]
This looks good to me, well done!

**Comment by ibro45** [comment]
Thanks! 
> @ibro45 You can squash-merge it.

I can't do that as I have no write access.

**Pull request merged**